sormas-app/app/src/main/java/de/symeda/sormas/app/rest/RetroProvider.java

```diff
@@ -308,7 +309,10 @@ public static Integer getNumberOfEntitiesToBePulledInOneBatch(long approximateJs
 		double compressedJsonSizeInBits = approximateJsonSizeInBytes * 8 / JSON_COMPRESSION_FACTOR;
 		int batchSize =
 			Math.toIntExact(Math.round(getNetworkDownloadSpeedInKbps(context) * ASSUMED_TRANSFER_TIME_IN_SECONDS * 1024 / compressedJsonSizeInBits));
-		return batchSize < 10 ? 10 : batchSize;
+		// Restrict the batch size to a maximum value to avoid backend queries leading to a timeout
+		batchSize = Math.min(MAX_BATCH_SIZE, Math.max(10, batchSize));
+		batchSize = (int) (10 * Math.sqrt(batchSize / 10f));
+		return batchSize;
```

Comment:
Limiting the batch size to 500 has to be done last!

Reply:
True, no idea why I wrote it like that :D